blog/content/post/2023-09-27-aem_phase2.md

```diff
@@ -0,0 +1,158 @@
+---
+title: TrenchBoot Anti Evil Maid - Phase 2
+abstract: 'TrenchBoot Anti Evil Maid project for Qubes OS is progressing. With
+           the addition of TPM 2.0 support, Anti Evil Maid gains much higher
+           adoption and possibilities than ever before.'
+cover: /covers/trenchboot-logo.png
+author: michal.zygowski
+layout: post
+published: true
+date: 2023-09-27
+archives: "2023"
+
+tags:
+  - coreboot
+  - QubesOS
+  - TrenchBoot
+  - GRUB2
+  - Xen
+  - Hypervisor
+categories:
+  - Firmware
+  - Bootloader
+  - Hypervisor
+  - OS Dev
+  - Security
+
+
+---
+
+# Introduction
+
+If you haven't read the previous [TrenchBoot AEM blog
+post](https://blog.3mdeb.com/2023/2023-01-31-trenchboot-aem-for-qubesos/) or
+our [project plan](https://docs.dasharo.com/projects/trenchboot-aem-v2/) yet,
+it may be a good lecture to catch up on what the TrenchBoot AEM project is
+about. This post will guide you through the most recent progress and show a
+demo of TrenchBoot AEM in action, plus installation steps.
+
+## What's new in TrenchBoot AEM?
+
+The main focus of this phase was to get the TPM 2.0 working with Qubes OS AEM
+scripts on the DRTM boot flow in TrenchBoot (GRUB and Xen). Let's go through
+each of the tasks quickly.
+
+### Implement parallel CPU cores bring-up for DRTM launch
+
+It is not strictly related to TPM 2.0 support, but after phase 1, it occurred
+to be necessary to clean up the code for CPU bring-up in Xen. As the phase 1
+was just a Proof of Concept, some inefficiencies in the code happened to land.
+The most significant change in this [pull
+request](https://github.com/TrenchBoot/xen/pull/1/files) is the parallel
+bring-up of the CPUs, in opposite to the previous implementation, which caused
+the CPUs to wait in a queue. It had to be fixed because if an interrupt
+occurred during the wait time, it could pose a severe threat to security.
+
+### Support for TPM 2.0 module in Xen
+
+With the introduction of TPM 2.0 support, Xen also required changes that would
+perform the dom0 kernel and initrd measurements properly. As a quick reminder,
+TPM 1.2 and TPM 2.0 communication protocols to extend the PCRs with software
+measurements differ slightly. Also, TPM 2.0 uses stronger hash algorithms than
+TPM 1.2, like SHA256, which had to be added to Xen. Changes are outlined in
+this [pull request](https://github.com/TrenchBoot/xen/pull/3/files).
+
+### Support for TPM 2.0 event log in Xen
+
+Similarly to the previous task, besides the communication protocol difference,
+the event log, where the information about measurements is stored, differs
+between TPM 1.2 and TPM 2.0. As a next step after measuring the dom0 kernel
+and initrd to TPM 2.0 PCRs, we had to record the hashes in the event log in
+TPM 2.0 compliant format. The event log is a very important chunk of
+information, which lets one confirm what is measured and if the TPM PCRs
+actually match the values that have been extended (basically an attestation).
```

Comment:
False. You can't trust event log unless you know that final PCR values are correct. Without it, anything with event log matching final values would be considered as trusted, but at best it only means that the code producing the log is proper.

```suggestion
actually match the values that have been extended.
```